### PR TITLE
refactor: create sql parser spi and a druid type sql parser

### PR DESCRIPTION
**Pull request by ggndnn**
### Ⅰ. Describe what this PR did
This PR extracted sql parser part from rm-datasource to create sql parser SPI (seata-sqlparser-core), all imports from druid in rm-datasource were removed except unit test code. Druid related code was moved to a new project seata-sqlparser-druid where a druid type sql parser was implemented.

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java [hunk 1]
```diff
@@ -15,17 +15,12 @@
...
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Map.Entry;
-import java.util.Objects;
-
 import io.seata.common.exception.NotSupportYetException;
 import io.seata.common.util.CollectionUtils;
 
+import java.util.*;
```
Comment (jsbxyyx) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java]: not use *

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/oracle/OracleInsertRecognizer.java [hunk 2]
```diff
@@ -13,27 +13,22 @@
...
 
 import com.alibaba.druid.sql.ast.SQLExpr;
 import com.alibaba.druid.sql.ast.SQLStatement;
-import com.alibaba.druid.sql.ast.expr.SQLIdentifierExpr;
-import com.alibaba.druid.sql.ast.expr.SQLMethodInvokeExpr;
-import com.alibaba.druid.sql.ast.expr.SQLNullExpr;
-import com.alibaba.druid.sql.ast.expr.SQLSequenceExpr;
-import com.alibaba.druid.sql.ast.expr.SQLValuableExpr;
-import com.alibaba.druid.sql.ast.expr.SQLVariantRefExpr;
+import com.alibaba.druid.sql.ast.expr.*;
```
Comment (jsbxyyx) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/oracle/OracleInsertRecognizer.java]: not use *

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java (hunk 1 above)
Comment (ggndnn) [reply rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMeta.java]: fixed.

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/oracle/OracleInsertRecognizer.java (hunk 2 above)
Comment (ggndnn) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/oracle/OracleInsertRecognizer.java]: fixed.

**Comment by lovepoem** [comment]
@ggndnn please fix the conflict

**Comment by ggndnn** [comment]
> @ggndnn please fix the conflict

Ok, I will fix it.

**Comment by ggndnn** [comment]
@lovepoem fixed.

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 3]
```diff
@@ -425,4 +425,9 @@
      * The constant DISABLE_GLOBAL_TRANSACTION.
      */
     public static final String DISABLE_GLOBAL_TRANSACTION = SERVICE_PREFIX + "disableGlobalTransaction";
+
+    /**
+     * The constant SQL_PARSER_TYPE.
+     */
+    public static final String SQL_PARSER_TYPE = "sqlparser.type";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: suggest replace "sqlparser.type" with CLIENT_RM_PREFIX+"sqlparser.type", and please add it in script module.

File: rm-datasource/src/main/java/io/seata/rm/datasource/sql/SQLVisitorFactory.java [hunk 4]
```diff
@@ -15,20 +15,25 @@
...
+ * @author ggndnn
  */
 public class SQLVisitorFactory {
+    /**
+     * SQLRecognizerFactory.
+     */
+    private final static SQLRecognizerFactory SQL_RECOGNIZER_FACTORY;
+
+    static {
+        String sqlparserType = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.SQL_PARSER_TYPE, "druid");
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/sql/SQLVisitorFactory.java]: SPI name and here need to define constants.

**Pull request merged**